File: C/2.c
#include<stdio.h>

int main()
{
	int sum = 0;
	int temp;
	int prevFib = 1; // Previous Fibonacci number
	int curFib = 2; // Current Fibonacci number

	while(curFib < 4000000) {
		if (curFib % 2 == 0) {
			sum += curFib;
		}
		temp = prevFib;
		prevFib = curFib;
		curFib += temp;
	}
	printf("Sum of Fib numbers: %d\n", sum);
	return 0;
}

File: Python/2.py
# Problem 2:
# By considering the terms in the Fibonacci sequence whose values do not exceed four million, find the sum of the even-valued terms.

# Approach:
# Instead of using recursion, use iteration. (Faster) 
# Keep track of two previous fibonacci numbers. 

sum_fib = 0

prev_fib = 1
cur_fib = 2

while cur_fib < 4000000:
  if cur_fib % 2 == 0:
    sum_fib += cur_fib
  temp = cur_fib
  cur_fib = cur_fib + prev_fib
  prev_fib = temp

print sum_fib